### PR TITLE
community: Add instructions in InfinityEmbeddings

### PR DESCRIPTION
**Pull request by baptiste-pasquier**
**Description:**

Add instructions for embedding with Infinity:

```python
embed_instruction: str = ""
query_instruction: str = ""
```

Allows to pass instructions like "passage: " or "query: ". Defaults to "" (no instructions)

### CODE REVIEW

**Comment by baptiste-pasquier** [comment]
What do you think @michaelfeil ?

File: libs/community/langchain_community/embeddings/infinity.py [hunk 1]
```diff
@@ -43,6 +43,11 @@ class InfinityEmbeddings(BaseModel, Embeddings):
     client: Any = None  #: :meta private:
     """Infinity client."""
 
+    embed_instruction: str = "passage: "
```
Comment (michaelfeil) [inline libs/community/langchain_community/embeddings/infinity.py:46]: Should default to ""

**Review by michaelfeil — APPROVED** [review]
approved with changes

File: libs/community/langchain_community/embeddings/infinity.py (hunk 1 above)
Comment (baptiste-pasquier) [reply libs/community/langchain_community/embeddings/infinity.py:46]: default to "" for document and also query ?

Comment (baptiste-pasquier) [reply libs/community/langchain_community/embeddings/infinity.py:46]: For BGE, the instruction for query is `Represent this sentence for searching relevant passages:` 
https://huggingface.co/BAAI/bge-large-en-v1.5

Comment (michaelfeil) [reply libs/community/langchain_community/embeddings/infinity.py:46]: Incorrect, see their repo: `update embedding model: release bge-*-v1.5 embedding model to alleviate the issue of the similarity distribution, and enhance its retrieval ability without instruction.` - by adding random instruction very deep in langchain, users of langchain will become frustrated, as they do not understand why their retrieval accuracy dropped. 

Please remove both instructions, thank you.

Comment (baptiste-pasquier) [reply libs/community/langchain_community/embeddings/infinity.py]: Thanks for the explanation, I set the defaults to ""

**Comment by efriis** [comment]
this is better done externally when you have the documents! This implementation will also prepend the instructions many times if you call the query functions I think

**Pull request closed without merging**